### DISCUSSION TITLE
[proposal] Better extension UX

### DISCUSSION THREAD

**Discussion by d1y** [zed-industries/zed#12455] [Show and tell]
I hope to improve the user experience of zed extension

## zed-cli

Ability to operate extension related operations in the cli tool

```sh
# Install extension
zed install extension-name
# Short
zed i extension-name

# Reinstall and focus
zed reinstall -f extension-name

# List installed extensions
zed ls-extensions

# Search extensions
zed search-extension extension-name

# Uninstall extension
zed uninstall extension-name

# Show extension information (metadata)
zed info-extension extension-name

```
## zed-scheme

Add a similar to` zed://extension/git-firefly `
This path protocol allows users to redirect to extension store and install (or preview)
And design a badge to display and link in the README of a third-party warehouse

**Comment by Moshyfawn** [comment]
FYI For the CLI part, there is already a proposal/feature request at

- #12324

For the path protocol.. I just remembered that I actually suggested something similar a while ago at

- #10794

**Reply by Moshyfawn** [reply]
This is just to say that these might already be the conversations you can work off of / collaborate with / link your work to.
